ui/pages/settings/flask/view-snap/view-snap.js

```diff
@@ -109,138 +108,142 @@ function ViewSnap() {
+        </Text>
+        <Box marginTop={4}>
+          <Tooltip interactive position="bottom" html={t('snapsToggle')}>
+            <ToggleButton
+              value={snap.enabled}
+              onToggle={onToggle}
+              className="view-snap__toggle-button"
+            />
+          </Tooltip>
         </Box>
-        <Box
-          className="view-snap__content-container"
-          width={FRACTIONS.SEVEN_TWELFTHS}
-        >
-          <div className="view-snap__section">
-            <Typography
-              variant={TypographyVariant.H6}
-              color={TextColor.textAlternative}
-              boxProps={{ marginTop: 5 }}
-            >
-              {snap.manifest.description}
-            </Typography>
-          </div>
-          <div className="view-snap__section view-snap__permission-list">
-            <Typography variant={TypographyVariant.H4}>
-              {t('permissions')}
-            </Typography>
-            <Typography
-              variant={TypographyVariant.H6}
-              color={TextColor.textAlternative}
-            >
-              {t('snapAccess', [snap.manifest.proposedName])}
-            </Typography>
-            <Box width={BLOCK_SIZES.FULL}>
-              <PermissionsConnectPermissionList
-                permissions={permissions ?? {}}
-                targetSubjectMetadata={targetSubjectMetadata}
-              />
-            </Box>
-          </div>
-          <div className="view-snap__section">
-            <Box width="11/12">
-              <Typography variant={TypographyVariant.H4}>
-                {t('connectedSites')}
-              </Typography>
-              <Typography
-                variant={TypographyVariant.H6}
-                color={TextColor.textAlternative}
-              >
-                {t('connectedSnapSites', [snap.manifest.proposedName])}
-              </Typography>
-              <ConnectedSitesList
-                connectedSubjects={connectedSubjects}
-                onDisconnect={(origin) => {
-                  onDisconnect(origin, snap.id);
-                }}
-              />
-            </Box>
-          </div>
-          <div className="view-snap__section">
-            <Typography variant={TypographyVariant.H4}>
-              {t('removeSnap')}
-            </Typography>
-            <Typography
-              variant={TypographyVariant.H6}
-              color={TextColor.textAlternative}
-              boxProps={{ paddingBottom: 3 }}
-            >
-              {t('removeSnapDescription')}
-            </Typography>
-            <Button
-              className="view-snap__remove-button"
-              type="danger"
-              css={{
-                maxWidth: '175px',
+      </Box>
+      <Box className="view-snap__permissions" marginTop={12}>
+        <Text variant={TextVariant.bodyLgMedium} marginLeft={4} marginRight={4}>
+          {t('permissions')}
+        </Text>
+        <PermissionsConnectPermissionList
```

Comment:
Replace this with `SnapPermissionsList` when https://github.com/MetaMask/metamask-extension/pull/18470 is merged.